[v2.7] k8s November 2023 patches

vardhaman22 opened the pull request:
**since november patches are released, skipping the october patches.**
related issues: https://github.com/rancher/rancher/issues/43073, https://github.com/rancher/rancher/issues/43152, https://github.com/rancher/rancher/issues/43205,
https://github.com/rancher/rancher/issues/43638, https://github.com/rancher/rancher/issues/43640, https://github.com/rancher/rancher/issues/43213

- adds k8s v1.25.16, v1.26.11 versions for rke1
- for all the new versions uses etcd v3.5.9 and rke-tools v0.1.96
- adds nginx v1.9.4 template and use it in all newly introduced versions.
- for nginx v1.9.4 update ingress-nginx/kube-webhook-certgen image to v20231011-8b53cabe0 [ref](https://github.com/kubernetes/ingress-nginx/blob/5583f90c7f3525b6267747d6b07541c01218a9ea/deploy/static/provider/cloud/deploy.yaml#L591)
- also includes the latest changes of dev-v2.7-2023-09-patches branch (needed because a few more changes went in for september patches)
- bumped calico to v3.26.3 for new k8s versions
- bumped k8s dns images to 1.22.28 for new k8s versions
- included hyperkube-base fix https://github.com/rancher/rancher/issues/43213

~Also, covers https://github.com/rancher/rancher/issues/43002~ Edit: It doesn't because destination branch is not `dev-v2.7`

cwayne18 committed:
Merge pull request #1239 from rancher/dev-v2.7

KDM Release for Rancher v2.7.7

brooksn committed:
v1.25.15+rke2r1 v1.26.10+rke2r1

Signed-off-by: Brooks Newberry <brooks@newberry.com>

brooksn committed:
remove RKE2 sep patches

Signed-off-by: Brooks Newberry <brooks@newberry.com>

vardhaman22 committed:
added nginx v1.9.4 template

superseb committed:
Bump rke-tools to v0.1.96

superseb committed:
go generate

superseb committed:
Merge pull request #1259 from superseb/v27202309patch_update_rketools_v0196

[dev-v2.7-2023-09-patches] Update rke-tools to v0.1.96

Commit:
[dev-v2.7] K3s patch release October (#1252)

* update v1.26.10-rc1+k3s1 and v1.25.15-rc1+k3s1

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* remove v1.26.9+k3s1 and v1.25.14+k3s1

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* update to v1.26.10-rc2+k3s1 and v1.25.15-rc2+k3s1

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* update to v1.26.10+k3s1 and v1.25.15+k3s1

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

---------

Signed-off-by: Johnatas <johnatasr@hotmail.com>

brooksn committed:
go generate

Signed-off-by: Brooks Newberry <brooks@newberry.com>

brooksn committed:
Merge pull request #1253 from brooksn/rke2-patch-2023-oct-2.7

[dev-v2.7] RKE2 patch releases

brooksn committed:
v1.25.15+rke2r2 v1.26.10+rke2r2

Signed-off-by: Brooks Newberry <brooks@newberry.com>

brooksn committed:
go generate

Signed-off-by: Brooks Newberry <brooks@newberry.com>

johnatasr committed:
[dev-v2.7] K3s2 patch release (#1264)

* update to v1.26.10-rc1+k3s2 and v1.25.15-rc1+k3s2

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* update to v1.26.10-rc2+k3s2 and v1.25.15-rc2+k3s2

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* update to v1.26.10-rc3+k3s2 and v1.25.15-rc3+k3s2

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* update to v1.26.10+k3s2 and v1.25.15+k3s2

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

---------

Signed-off-by: Johnatas <johnatasr@hotmail.com>

brooksn committed:
Merge pull request #1265 from brooksn/rke2-oct-r2-v2.7

[v2.7] RKE2 OOB October patch release

snasovich reviewed (changes requested):
@vardhaman22 , please update `rke-tools` to `v0.1.96`. While at it, could you please also update it in previous versions as done per https://github.com/rancher/kontainer-driver-metadata/pull/1259 (in fact, please bring in other changes that got into `dev-v2.7-2023-09-patches` branch after we branched for `dev-v2.7-2023-10-patches` - e.g. RKE2/k3s R2 releases).

vardhaman22 commented:
> @vardhaman22 , please update `rke-tools` to `v0.1.96`. While at it, could you please also update it in previous versions as done per #1259 (in fact, please bring in other changes that got into `dev-v2.7-2023-09-patches` branch after we branched for `dev-v2.7-2023-10-patches` - e.g. RKE2/k3s R2 releases).

rebased my branch with  `dev-v2.7-2023-09-patches`  and also updated rke-tools image to v0.1.96 in the latest changes.

kinarashah committed:
Merge pull request #1266 from rancher/dev-v2.7-2023-09-patches

[v2.7] OOB KDM release for v2.7.x

vardhaman22 committed:
k8s november 2023 patches

vardhaman22 committed:
bump calico to v3.26.3 for k8s nov patches

vardhaman22 committed:
bump k8s dns images to 1.22.28 for nov 2023 k8s patches

vardhaman22 committed:
bump hyperkube image for nov 2023 patches

vardhaman22 committed:
updated rke default k8s version

vardhaman22 committed:
go generate

kinarashah commented on pkg/rke/templates/calico_v3.36.go:
file name: calico_v3.26.go

vardhaman22 replied:
updated

kinarashah reviewed (dismissed):
- confirmed the k8s system images are same as v2.8 PR 
- confirmed default k8s version 
- confirmed addon versions in templates.go

snasovich commented:
LGTM. From what I can see this includes ensuring we have changes that were released per Sep patches into this branch too.

Also, I have removed references to 2.8 issues in the description of this PR.

snasovich reviewed (approved):
LGTM, thank you for addressing https://github.com/rancher/kontainer-driver-metadata/pull/1256#discussion_r1416611256
